### ISSUE TITLE
Jellyfin performs a full transcode on SecondaryAudioNotSupported when using AAC codec

### ISSUE THREAD

**Issue by dferrg** [jellyfin/jellyfin#7649]
I'm using firefox to play a video file with multiple audio tracks. The first track plays well, but when I change to the others, jellyfin is transcoding the full file instead of just demuxing. It doesn't happen with MP3 audio, even with exactly the same video codec and container format.

**Media Info of the file**
Audio
Title Und - AAC - 6 ch - Default
Language und
Codec AAC
Codec tag mp4a
Profile LC
Channels 6 ch
Bitrate 224 kbps
Sample rate 48000 Hz
Default Yes

Audio
Title AAC - 6 ch
Codec AAC
Codec tag mp4a
Profile LC
Channels 6 ch
Bitrate 224 kbps
Sample rate 48000 Hz
Default No

Video
Title1080p H264 SDR
Codec H264
Codec tag avc1
AVC No
Profile High
Level 40
Resolution 1920x1040
Aspect ratio 1.85:1
Interlaced No
Framerate 23.97589
Bitrate 1644 kbps
Bit depth 8 bit
Video range SDR
Pixel format yuv420p
Ref frames 1
NAL 0

Audio
Title AAC - 6 ch
Codec AAC
Codec tag mp4a
Profile LC
Channels 6 ch
Bitrate 224 kbps
Sample rate 48000 Hz
Default No

**Logs**

[12:49:42] [INF] [25] Jellyfin.Api.Helpers.MediaInfoHelper: User policy for dani. EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[12:49:42] [INF] [25] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: Unknown Profile, Path: /data/movies/El.viatge.de.Chihiro(1080p.x264.ACC)(Cat.Esp.Eng).m4v, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[12:49:42] [INF] [25] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: VideoAudioCodecProfile, DirectPlay=false. Reason=Unknown Profile.IsSecondaryAudio Condition: Equals. ConditionValue: false. IsRequired: False. Path: /data/movies/El.viatge.de.Chihiro(1080p.x264.ACC)(Cat.Esp.Eng).m4v
[12:49:43] [INF] [110] Jellyfin.Api.Controllers.DynamicHlsController: Current HLS implementation doesn't support non-keyframe breaks but one is requested, ignoring that request
[12:49:43] [INF] [110] Jellyfin.Api.Helpers.TranscodingJobHelper: /usr/lib/jellyfin-ffmpeg/ffmpeg -f mov,mp4,m4a,3gp,3g2,mj2 -i file:"/data/movies/El.viatge.de.Chihiro(1080p.x264.ACC)(Cat.Esp.Eng).m4v" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:2 -map 0:3 -map -0:s -codec:v:0 libx264 -pix_fmt yuv420p -preset veryfast -crf 23 -maxrate 4112095 -bufsize 8224190 -profile:v:0 high -level 41 -x264opts:0 subme=0:me_range=4:rc_lookahead=10:me=dia:no_chroma_me:8x8dct=0:partitions=none  -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "scale=trunc(min(max(iw\,ih*dar)\,min(1920\,1040*dar))/2)*2:trunc(min(max(iw/dar\,ih)\,min(1920/dar\,1040))/2)*2" -start_at_zero -vsync -1 -codec:a:0 aac -ac 6 -ab 224424 -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 3 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/config/data/transcodes/51b2683d380fd7b31e67c074d9b8a0ff%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/data/transcodes/51b2683d380fd7b31e67c074d9b8a0ff.m3u8"

**Stats for Nerds Screenshots**
![image](https://user-images.githubusercontent.com/35787927/164892326-6e746a2b-af40-41fe-a42f-b9212d261d1b.png)

**Server System (please complete the following information):**
 - OS: Docker on Linux
 - Jellyfin Version: 10.7.7
 - Hardware settings & device: Raspberry Pi 4
 - Reverse proxy: nginx
 
**Client System (please complete the following information):**
 - Device: Laptop, i7 11th gen, GTX 1650
 - OS: Linux Mint
 - Client: Web browser
 - Browser (if Web client): Firefox
 - Client and Browser Version: 99.0

**Comment by cvium** [comment]
Have you tried the 10.8 beta?

**Comment by dferrg** [comment]
No, I don't. I was planning to transcode all my library to a common format, so it doesn't matter too much whether it's AAC or MP3 for me.
Is it supposed to be solved in 10.8? I can give it a try and give feedback if you want.

**Comment by Shadowghost** [comment]
10.8 supports partial transcodes so only the streams not compatible with your client are getting transcoded

**Comment by dferrg** [comment]
The streams themselves should be compatible since the primary track plays without transcoding. Also, the same video but transcoded to MP3 gets remuxed instead of transcoded. 

Though obviously if AAC can't be remuxed for some reason, transcoding just audio would be much better than transcoding the entire file.

**Comment by whiteowl3** [comment]
please tag 10.7.7

**Comment by Traace** [comment]
Jellyfin sill performs a fill transcode on live-tv for me. The Media plays fine when I disable transcoding, so there shouldn't be any need for it.

**Comment by 300EMI** [comment]
Hi,

I have the same problem: SecondaryAudioNotSupported

Video
Title: 1080p H264 SDR
Codec: H264

Audio
Title: Deutsch 2.0 - AAC - Stereo - Default
Language: deu
Codec: AAC
Codec tag: mp4a

Audio
Title: Deutsch 5.1 - Dolby Digital
Language: deu
Codec: AC3

Audio
Title: Englisch 2.0 - AAC - Stereo
Language: eng
Codec: AAC

If I play the movie with the default/ first soundtrack, playback starts without “transcoding”. (Windows 11, Firefox, Edge and Chrome) If I play the third soundtrack (TranscodeReasons=SecondaryAudioNotSupported) and the Jellyfin starts to “transcode” the stream



Server: jellyfin
Version: 10.8.9
Operating System: Linux
Architecture: X64

Client:
Windows 10
Firefox 111.0 (64-Bit)
Chrome Version 111.0.5563.65 (Offizieller Build) (64-Bit)
Microsoft Edge: Version 111.0.1661.44 (Offizielles Build) (64-Bit)

BR
René

**Comment by xescure** [comment]
I'm still encountering this issue and nothing was done to remedy the problem. Can this be reopened?

**Comment by dancingmadkefka** [comment]
I am having the same issue. I have a 10GB episode with two tracks: both E-AC-3 JOC. 
I played it on my LG OLED B8 Jellyfin native app, and it said "Remuxing due to container" or something. MKV container, should not be an issue. Then I turned on HWA and selected all the codecs and enabled HEVC transcoding and it started giving me the above issue. It also stutters a lot despite the server not being overloaded and my network being uncongested.
